### ISSUE TITLE
Doesn't boot on XCP-NG/XEN

### ISSUE THREAD

**Issue by timemaster5** [vmware/photon#1555]
### Describe the bug

The installer successfully starts but fails as it doesn't see the mounted ISO.

When I use OVA on Xcp-ng, it doesn't boot at all. I can only see the boot menu and then nothing, a black screen with a blinking cursor.

I know XP-ng is not a target platform for this OS, but in hybrid environments, this OS can't be moved from one hypervisor to another, as I tested it.

### Reproduction steps

1. Create a new VM in xcp-ng with x86_64 minimal/full ISO
2. Import OVA of x86_64 to xcp-ng

### Expected behavior

I would expect this system to boot on any hypervisor.

**Comment by dcasota** [comment]
With Photon OS 5, see https://vmware.github.io/photon/docs-v5/troubleshooting-guide/photon-installation-issue/, ide interface for cdrom isn't supported anymore. Can you check the setting of the or specifiy a cdrom sata controller ?

**Comment by timemaster5** [comment]
I know, but there is no such option in xen. I mean I can’t choose between ide and scsi, if I am correct there is a proprietary blkdev driver in kernel for it.On 24. 6. 2024, at 13:34, dcasota ***@***.***> wrote:﻿
With Photon OS 5, see https://vmware.github.io/photon/docs-v5/troubleshooting-guide/photon-installation-issue/, ide interface for cdrom isn't supported anymore. Can you check the setting of the or specifiy a cdrom sata controller ?

**Comment by dcasota** [comment]
I'm afraid, the official 5.0 ova and iso won't help in this case. Just some thoughts:
- Start with 4.0 (which still includes ide support) and upgrade to 5.0
- discuss with the Photon OS team your use case and a future pull request which includes the requirement
- make your custom build with specific .config parameters and modules

**Comment by timemaster5** [comment]
Yep, you're right. The official way doesn't work. That's why I am mentioning this. It might be actually useful to support other hypervisors too, as at this point we can't use them for datacenter migration and so on.

Do you think upgrading from 4.0 to 5.0 will work? I guess the kernel will be upgraded too, and the ide support is lost.

Custom build should work fine of course.

**Comment by vbrahmajosyula1** [comment]
We restored IDE support with this commit in photon 5, unfortunately the release iso does not have it.

https://github.com/vmware/photon/commit/72f6efefc3b597ed76ff5b317ce74abf106cb513

1. Upgrading from photon 4 to 5 is a reasonable approach.
2. You can also generate a new iso using latest packages in photon repo and it should work for you.
     - copy the RPMs from https://packages.vmware.com/photon/5.0/photon_updates_5.0_x86_64/ to your photon/stage/RPMS and run make iso.

**Comment by timemaster5** [comment]
Oh nice, I will try it ! thanks

**Comment by timemaster5** [comment]
I confirm I can install with 4.0 ISO and upgrade to 5.0 now successfully!

Thank you
